modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/portlet/action/ExportRootModelObjectDefinitionsMVCResourceCommand.java

```diff
@@ -0,0 +1,128 @@
+import com.liferay.portal.kernel.util.ContentTypes;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Time;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.vulcan.pagination.Page;
+
+import java.util.Collection;
+
+import javax.portlet.ResourceRequest;
+import javax.portlet.ResourceResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Marco Leo
+ * @author Gabriel Albuquerque
+ */
+@Component(
+	property = {
+		"javax.portlet.name=" + ObjectPortletKeys.OBJECT_DEFINITIONS,
+		"mvc.command.name=/object_definitions/export_root_model_object_definitions"
+	},
+	service = MVCResourceCommand.class
+)
+public class ExportRootModelObjectDefinitionsMVCResourceCommand
+	extends BaseMVCResourceCommand {
+
+	@Override
+	protected void doServeResource(
+			ResourceRequest resourceRequest, ResourceResponse resourceResponse)
+		throws Exception {
+
+		ObjectDefinitionResource.Builder builder =
+			_objectDefinitionResourceFactory.create();
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)resourceRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		ObjectDefinitionResource objectDefinitionResource = builder.user(
+			themeDisplay.getUser()
+		).build();
+
+		long objectDefinitionId = ParamUtil.getLong(
+			resourceRequest, "objectDefinitionId");
+
+		ObjectDefinition rootObjectDefinition =
+			objectDefinitionResource.getObjectDefinition(objectDefinitionId);
+
+		Page<ObjectDefinition> rootModelObjectDefinitionsPage =
+			objectDefinitionResource.getObjectDefinitionsPage(
+				null, null,
+				objectDefinitionResource.toFilter(
+					StringBundler.concat(
+						"rootObjectDefinitionExternalReferenceCode eq '",
+						rootObjectDefinition.getExternalReferenceCode(), "'")),
+				null, null);
+
+		Collection<ObjectDefinition> rootModelObjectDefinitions =
+			rootModelObjectDefinitionsPage.getItems();
+
+		JSONArray rootModelObjectDefinitionsJSONArray =
+			_jsonFactory.createJSONArray();
+
+		for (ObjectDefinition objectDefinition : rootModelObjectDefinitions) {
+			ExportImportObjectDefinitionUtil.prepareObjectDefinitionForExport(
+				_jsonFactory, objectDefinition);
+
+			JSONObject objectDefinitionJSONObject =
+				_jsonFactory.createJSONObject(objectDefinition.toString());
+
+			if (!FeatureFlagManagerUtil.isEnabled("LPS-135430")) {
+				objectDefinitionJSONObject.remove("storageType");
+			}
+
+			JSONObjectSanitizerUtil.sanitize(
+				objectDefinitionJSONObject,
+				new String[] {
+					"dateCreated", "dateModified", "id", "listTypeDefinitionId",
+					"objectDefinitionId", "objectDefinitionId1",
+					"objectDefinitionId2", "objectFieldId",
+					"objectRelationshipId", "titleObjectFieldId"
+				});
+
+			rootModelObjectDefinitionsJSONArray.put(objectDefinitionJSONObject);
+		}
+
+		String rootModelObjectDefinitionsJSONString =
+			rootModelObjectDefinitionsJSONArray.toString();
+
```

Comment:
Hey, I think you forgot rootModelObjectDefinitionsJSON**Array**String. Can you rename it please?